kebechet/managers/thoth_advise/thoth_advise.py

```diff
@@ -242,6 +284,15 @@ def run(self, labels: list, analysis_id=None):
                 branch = self.repo.git.checkout("-B", branch_name)  # noqa F841
                 self._cached_merge_requests = self.project.get_pr_list()
 
+                # run results needs to pass runtime_environment to Kebechet. This should always be present. It may
+                # actually be better to raise an error here. WDYT?
```

Comment:
The current schema restricts `runtime_environments` to have at least one entry so this should be fine:

https://github.com/thoth-station/thamos/blob/f7971f5acab00dd76ee01fe44aed1be82c3fa861/thamos/config.py#L119-L122

Long-term, it might be better to operate on the config instance from thamos - it will make sure the schema is valid and any config related checks can go there. kebechet can wrap calls to try-except and handle any config related issues at one place (e.g. open an issue to repo).

Reply:
I think we are not on the same page. `self.runtime_environment` refers to the runtime_environment name which is passed to Kebechet. This comment refers to the fact that Kebechet should always call advise with runtime environment set and the advise workflow should always schedule kebechet-run-results with the same runtime environment.  run-results theoretically should never be run without being passed a runtime_environment name.  I just want the behavior to be well defined here and was wondering if we should just default to the first entry or if we should raise an error.

Reply:
Ah, now I see - thanks for the explanation. I think being explicit here is the way to go, not to silently default. Can there be a case when the runtime environment name is changed while serving this request?

BTW using thamos' config abstraction can be still good.

Reply:
Yeah, the user could update the runtime environment after creating a request. I believe then we will fail when writing results. I think this is the behavior we want.